src/catalog/src/durable/persist.rs

```diff
@@ -998,16 +989,12 @@ impl UnopenedPersistCatalogState {
         // If this is `None`, no version was found in the upgrade shard. This is a brand-new
         // environment, and we don't need to worry about fencing existing users.
         if let Some(version_in_upgrade_shard) = version_in_upgrade_shard {
-            // IMPORTANT: We swap the order of arguments here! Normally it's
-            // `code_version, data_version`, and we check whether a given code
-            // version, which is usually _older_, is allowed to touch a shard
-            // that has been touched by a _future_ version.
-            //
-            // By inverting argument order, we check if our version is too far
-            // ahead of the version in the shard.
-            if mz_persist_client::cfg::check_data_version(&version_in_upgrade_shard, &version)
-                .is_err()
-            {
+            // Check that the current version of the code can handle data from the shard.
+            // (We used to reverse this check, to confirm that whatever code wrote the data
+            // in the shard would be able to read data written by the current version... but
+            // we now require the current code to be able to maintain compat with whatever
+            // data format versions pass this check.)
```

Comment:
Not a strong opinion, but this historical bit seems more confusing than helpful. It doesn't help understanding the current behavior, so why burden the reader with the knowledge?

(It _is_ helpful for this code review though!)